homework/src/app.js

```diff
@@ -1,11 +1,254 @@
+x === y; // false because as x==y then they can never be x===y
+z === y; // true, because they reference the same objects
+z == x; // false, because they reference different objects
+
+// Consider the following snippet:
+
+const o1 = { foo: 'bar' };
+const o2 = { foo: 'bar' };
+const o3 = o2;
+
+// Show that changing o2 changes o3 (or not) and changing o1 changes o3(or not).
+o2.key = 'changing';
+
+o2; //it would be {foo: 'bar', key: 'changing'}
+o3; // it would also be  {foo: 'bar', key: 'changing'}
+
+// changing o1 does not change o3 because they are not == ;
+
+// Does the order that you assign (o3 = o2 or o2 = o3) matter?
+// Answer: the assign order matters for the above example if we say const o3=o2; it would give us an error because we hav not defined o2 yet.
+
+// THE SECOND PART OF PREVIOUS ASSIGNMENT
+
+// 8.1
+
+let myFavoriteAnimals = ['lions', 'horses', 'eagles'];
+const myFavoriteTennisPlayer = 'Nadal';
+let myAge = 30;
+var answer = false;
+
+// 8.2
+
+console.log('the value of my variable myFavoriteAnimals is:' + myFavoriteAnimals);
+console.log('the value of my variable myFavoriteTennisPlayer  is:' + myFavoriteTennisPlayer);
+console.log('the value of my variable myAge is:' + myAge);
+console.log('the value of my variable answer is:' + answer);
+
+// 8.3
+
+console.log(" I think the type of my variable 'myFavoriteAnimals' is: 'object' ");
+console.log(" I think the type of my variable 'myFavoriteTennisPlayer'  is: 'string' ");
+console.log(" I think the type of my variable 'myAge' is: 'number' ");
+console.log(" I think the type of my variable 'answer' is: 'boolean' ");
+
+// 8.4
+
+console.log(typeof myFavoriteAnimals);
+
+console.log(typeof myFavoriteTennisPlayer);
+
+console.log(typeof myAge);
+
+console.log(typeof answer);
+
+// // 8.5  Now compare the types of your different variables with one another.
+// object ; String; Number; Boolean;
+
+// 8.6
+
+if (typeof myFavoriteAnimals == typeof myFavoriteTennisPlayer) {
+  console.log('Same Type');
+} else typeof myFavoriteAnimals != typeof myFavoriteTennisPlayer;
+{
+  console.log('Different Type');
+}
+
+if (typeof myFavoriteAnimals == typeof myAge) {
+  console.log('Same Type');
+} else typeof myFavoriteAnimals != typeof myAge;
+{
+  console.log('Different Type');
+}
+
+if (typeof myFavoriteAnimals == typeof answer) {
+  console.log('Same Type');
+} else typeof myFavoriteAnimals != typeof answer;
+{
+  console.log('Different Type');
+}
+
+if (typeof myFavoriteTennisPlayer == typeof myAge) {
+  console.log('Same Type');
+} else typeof myFavoriteTennisPlayer != typeof myAge;
+{
+  console.log('Different Type');
+}
+
+if (typeof myFavoriteTennisPlayer == typeof myAge) {
+  console.log('Same Type');
+} else typeof myFavoriteTennisPlayer != typeof myAge;
+{
+  console.log('Different Type');
+}
+
+if (typeof myAge == typeof answer) {
+  console.log('Same Type');
+} else typeof myAge != typeof answer;
+{
+  console.log('Different Type');
+}
```

Comment:
Ah - try to avoid using == operator while equating two values. Always try to use === as it is more reliable. Try and go through this documentation please - https://developer.mozilla.org/nl/docs/Web/JavaScript/Equality_comparisons_and_sameness